packages/file_selector/file_selector_platform_interface/lib/src/types/x_file/html.dart

```diff
@@ -0,0 +1,131 @@
+import 'dart:convert';
+import 'dart:typed_data';
+
+import 'package:http/http.dart' as http show readBytes;
+import 'package:meta/meta.dart';
+import 'dart:html';
+
+import './base.dart';
+
+import '../x_type/x_type.dart';
+
+/// A XFile that works on web.
+///
+/// It wraps the bytes of a selected file.
+class XFile extends XFileBase {
+  String path;
+  final XType type;
+
+  final Uint8List _data;
+  final int _length;
+  final String name;
+  Element _target;
+
+  /// Construct a XFile object from its ObjectUrl.
+  ///
+  /// Optionally, this can be initialized with `bytes` and `length`
+  /// so no http requests are performed to retrieve files later.
+  ///
+  /// `name` needs to be passed from the outside, since we only have
+  /// access to it while we create the ObjectUrl.
+  XFile(
+      this.path, {
+        this.type,
+        this.name,
+        int length,
+        Uint8List bytes,
+      })  : _data = bytes,
+        _length = length,
+        super(path) {
+    // Create a DOM container where we can host the anchor.
+    _target = _ensureInitialized(this.name + '-x-file-dom-element');
+  }
+
+  /// Construct an XFile from its data
+  XFile.fromData(
+      Uint8List bytes, {
+        this.type,
+        this.name,
+        int length,
+      })  : _data = bytes,
+        _length = length,
+        super('') {
+    Blob blob;
+    if (type.mime == null) {
+      blob = Blob([bytes]);
+    } else {
+      blob = Blob([bytes], type.mime);
+    }
+    this.path = Url.createObjectUrl(blob);
+    // Create a DOM container where we can host the anchor.
+    _target = _ensureInitialized(this.name + '-x-file-dom-element');
+  }
+
+
+
+  Future<Uint8List> get _bytes async {
+    if (_data != null) {
+      return Future.value(UnmodifiableUint8ListView(_data));
+    }
+    return http.readBytes(path);
+  }
+
+  @override
+  Future<int> length() async {
+    return _length ?? (await _bytes).length;
+  }
+
+  @override
+  Future<String> readAsString({Encoding encoding = utf8}) async {
+    return encoding.decode(await _bytes);
+  }
+
+  @override
+  Future<Uint8List> readAsBytes() async {
+    return Future.value(await _bytes);
+  }
+
+  @override
+  Stream<Uint8List> openRead([int start, int end]) async* {
```

Comment:
@jasonpanelli it would be awesome to see [support for streaming in the file data](https://joji.me/en-us/blog/processing-huge-files-using-filereader-readasarraybuffer-in-web-browser/) as opposed to reading it all into memory first here!

Reply:
This is inherited from an old implementation I did a while back, and I agree this'd be a great change. The plan for XFile is to eventually become its own package, so we'll be able to add these improvements later!